modules/core/shared/src/main/scala/lucuma/core/math/Coverage.scala

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.math
+
+import cats.Eq
+import cats.syntax.order._
+import coulomb._
+import eu.timepit.refined.types.numeric.{NonNegInt, PosInt}
+import lucuma.core.math.units._
+
+/** Wavelength coverage. */
+sealed trait Coverage {
+  import Coverage.{ Empty, Range }
+
+  /** Intersect this `Coverage` with another. */
+  def ⋂(other: Coverage): Coverage =
+    (this, other) match {
+      case (Empty, _) => Empty
+      case (_, Empty) => Empty
+      case (Range(a, b), Range(aʹ, bʹ)) => Coverage(a max aʹ, b min bʹ)
+    }
+
+  /** Coverage width; i.e., difference between max and min (or zero). */
+  def width: Quantity[NonNegInt, Picometer] =
+    this match {
+      case Empty       => NonNegInt.unsafeFrom(0).withUnit[Picometer]
+      case Range(a, b) => NonNegInt.unsafeFrom(b.toPicometers.value.value - a.toPicometers.value.value).withUnit[Picometer]
```

Comment:
Given there is a substraction, this could go negative or is that protected by the `Range` constructor?

Reply:
Right, `Range` can only be created if it max is strictly > min.